### PR TITLE
Search DomainEvent commons (migration from 2.2.X) to 3.X

### PR DESCRIPTION
**Pull request by EvgeniyMurenko**

### CODE REVIEW

**Review by major13ua — APPROVED** [review]

**Review by sergeysenja1992 — APPROVED** [review]

File: xm-commons-domainevent-outbox/src/main/java/com/icthh/xm/commons/domainevent/outbox/config/HibernateConfig.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import javax.sql.DataSource;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Configuration
+@EnableJpaRepositories(basePackageClasses = OutboxRepository.class)
+public class HibernateConfig {
```
[the remark this answers is not in the document]
Comment (EvgeniyMurenko) [reply xm-commons-domainevent-outbox/src/main/java/com/icthh/xm/commons/domainevent/outbox/config/HibernateConfig.java]: deleted

File: xm-commons-idp/src/main/java/com/icthh/xm/commons/domainevent/idp/IdpConfigUtils.java [hunk 2]
```diff
@@ -1,9 +1,9 @@
-package com.icthh.xm.commons.domain.idp;
```
Comment (amedvedchuk) [inline xm-commons-idp/src/main/java/com/icthh/xm/commons/domainevent/idp/IdpConfigUtils.java:1]: @EvgeniyMurenko why do you touch IDP feature related classes?

Comment (EvgeniyMurenko) [reply xm-commons-idp/src/main/java/com/icthh/xm/commons/domainevent/idp/IdpConfigUtils.java]: made a transfer using a patch from the "Outbox" branch.
rollback

File: xm-commons-domainevent-outbox/src/main/java/com/icthh/xm/commons/domainevent/outbox/config/HibernateConfig.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    private BeanFactory beanFactory;
+
+    @Bean
+    JpaVendorAdapter jpaVendorAdapter() {
+        return new HibernateJpaVendorAdapter();
+    }
+
+    @Bean
+    @Primary
+    LocalContainerEntityManagerFactoryBean entityManagerFactory(DataSource dataSource,
```
Comment (sergeysenja1992) [inline xm-commons-domainevent-outbox/src/main/java/com/icthh/xm/commons/domainevent/outbox/config/HibernateConfig.java:47]: why we use Primary and override LocalContainerEntityManagerFactoryBean from DatabaseConfiguration?

**Review by sergeysenja1992 — CHANGES_REQUESTED** [review]
Do not user Primary annotation and do not override LocalContainerEntityManagerFactoryBean

File: xm-commons-migration-db/src/main/java/com/icthh/xm/commons/migration/db/config/DatabaseConfiguration.java [hunk 4]
```diff
@@ -98,7 +101,8 @@ public LocalContainerEntityManagerFactoryBean entityManagerFactory(
             DataSource dataSource,
             MultiTenantConnectionProvider multiTenantConnectionProviderImpl,
             CurrentTenantIdentifierResolver currentTenantIdentifierResolverImpl,
-            LocalValidatorFactoryBean localValidatorFactoryBean) {
+            LocalValidatorFactoryBean localValidatorFactoryBean,
+            List<EntityScanPackageProvider> entityScanPackageProviderList) {
```
Comment (sergeysenja1992) [inline xm-commons-migration-db/src/main/java/com/icthh/xm/commons/migration/db/config/DatabaseConfiguration.java:105]: will it work when outbox not added to classpath and no EntityScanPackageProvider present?

**Review by sergeysenja1992 — APPROVED** [review]

**Review by pkev777 — APPROVED** [review]

**Pull request merged**